### PR TITLE
[AFC] Implemented Catti-brie of Mithral Hall

### PR DESCRIPTION
**Pull request by zeffirojoe**
#7978

### CODE REVIEW

File: Mage.Sets/src/mage/cards/c/CattiBrieOfMithralHall.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    public CattiBrieOfMithralHall copy() {
+        return new CattiBrieOfMithralHall(this);
+    }
+}
+
+class CattiBrieOfMithralHallEffect extends OneShotEffect {
+
+    CattiBrieOfMithralHallEffect() {
+        super(Outcome.Damage);
+        staticText = "Remove all +1/+1 counters from {this}, deal X damage to target attacking or blocking creature an opponent controls, where X is the number of counters removed this way.";
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/c/CattiBrieOfMithralHall.java:87]: Removing the counters is a cost, not part of the effect. Take a look at cards like Geometric Nexus and Relic Amulet for examples.

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/c/CattiBrieOfMithralHall.java (hunk 1 above)
Comment (zeffirojoe) [reply Mage.Sets/src/mage/cards/c/CattiBrieOfMithralHall.java:95]: Something like my recent commit?

Comment (theelk801) [reply Mage.Sets/src/mage/cards/c/CattiBrieOfMithralHall.java:95]: yeah, although you don't need to make a special `OneShotEffect` for as `DamageTargetEffect` can accept a `DynamicValue`

Comment (zeffirojoe) [reply Mage.Sets/src/mage/cards/c/CattiBrieOfMithralHall.java]: Understood. Updated to use `DamageTargetEffect`

**Review by theelk801 — APPROVED** [review]

**Pull request merged**